code/code/cmd/cmd_bash.cc

```diff
@@ -94,162 +93,216 @@ bool TBeing::canBash(TBeing *victim, silentTypeT silent)
+
+  // Without this check, many warrior mobs won't be able to bash
+  if (attacker->isPc()) {
+    /*
+      Allow bash with weapon + no shield after 10% in brawling
+    */
+    if (brawlingLearnedness <= 10 && !isHoldingShield) {    
+        attacker->sendTo("You're not skilled enough to bash without a shield!\n\r");
+        return FALSE;    
     }
-  } else if (skill == SKILL_BASH) {
-    // warrior bashing with disc_brawling > 10%
-    if (learning <= 25) 
-      shieldAdj = learning;
-    else 
-      shieldAdj = 25 + ((learning-25)/4);
-  } else {
-    if (((obj = c->heldInSecHand()) &&
-        (tbc = dynamic_cast<TBaseClothing *>(obj)) &&
-         tbc->isShield())) {
-        shieldWtBonus=(tbc->getWeight() / 5) + 1;
-    } else if (learning == MAX_DISC_LEARNEDNESS) {
-    } else {
-      c->sendTo("You are not skilled enough to bash without a shield!\n\r");
+
+    /* 
+      Bashing while dual wielding, or single wielding a one-handed weapon in either hand 
+      with no shield, requires even higher brawling disc as there's no shield or heavy 
+      weapon to assist with the bash 
+    */
+    if (
+      brawlingLearnedness < 50 && 
+      !isHoldingShield && 
+      !isWielding2Hander && 
+      (weaponInPrimaryHand || weaponInSecondaryHand)
+    ) {
+      attacker->sendTo("You're not skilled enough to bash with one-handed weapons!\n\r");
+      return FALSE; 
+    } 
+
+    /* 
+      Require near-maxed brawling disc to bash with no weapons or shield at all. Probably
+      not a super common scenario. 
+    */
+    if (
+      brawlingLearnedness < 90 && 
+      isBarehanded &&
+      !isHoldingShield
+    ) {
+      attacker->sendTo("You're not skilled enough to bash without a weapon!\n\r");
       return FALSE;
     }
   }
 
-  // high level warriors are able to abuse bash
-  // mob on ground, attacks/defends horribly and takes extra dam
-  // encourage warriors to use alternative more costly methods like
-  // grapple, and bslam
-  // bat 11/13/98
-#if 0
-  if (::number(0,99) < victim->GetMaxLevel()) {
-    act("You rush at $N, but $E steps backward avoiding your bash.",
-       FALSE, c, 0, victim, TO_CHAR);
-    act("$n rushes at you, but you step backward avoiding $s bash.",
-       FALSE, c, 0, victim, TO_VICT);
-    act("$n rushes at $N, but $E steps backward avoiding $s bash.",
-       FALSE, c, 0, victim, TO_NOTVICT);
-    // don't allow this to prevent a fight
-    c->reconcileDamage(victim, 0, skill);
-    // return true so the lag and stuff
+  /*
+    If attacker is not tanking, pass agility check to successfully get into 
+    position for a bash attempt. Makes bashing while not tanking more difficult
+    but still possible.
+  */
+  if (victim->fight() && victim->fight() != attacker && !attacker->isAgile(0)) {
+    attacker->sendTo("You try to line up a bash but can't find a good angle!\n\r");
+    // Still add skill lag, as with deikhan charge
```

Comment:
Changed it so that bashing someone who's fighting but that you're not actively tanking requires passing an agility check to "line up" the bash amidst the chaos. Was aiming for something similar to how Deikhan charge works.

I think these sorts of stat checks have a lot of potential throughout the codebase, but they're not used super often as is.